vsphere/tests/test_check.py

```diff
@@ -737,3 +737,56 @@ def test_no_infra_cache_no_perf_values(aggregator, realtime_instance, dd_run_che
         )
 
         aggregator.assert_all_metrics_covered()
+
+
+@pytest.mark.usefixtures("mock_type", "mock_threadpool", "mock_api", "mock_rest_api")
+def test_hostname_case(aggregator, realtime_instance, dd_run_check):
```

Comment:
nit- can you move this into `test_unit` and break it up into 3 separate tests or a parameterized test?

Reply:
Broke up this test into 3 parts and put in `test_unit.py`